packages/logger/test/end-to-end/scripts/run-loggers-with-test-case.js

```diff
@@ -25,3 +31,15 @@ if (nLogger[method]) {
 if (reliabilityKitLogger[method]) {
 	reliabilityKitLogger[method](...args, { _logger: 'reliabilityKit' });
 }
+
+// Run n-mask-logger
+if (nMaskLogger[method]) {
+	nMaskLogger[method](...args, { _logger: 'nextMaskLogger' });
+}
```

Comment:
I am unclear on what exactly this file is testing. It feels a bit strange to have the existing logger packages as dependencies to Reliability Kit and also have tests dependent on them, though I am probably missing something.

Can you fill me in please?

Reply:
Gladly! Everything in the `packages/logger/test/end-to-end` folder is designed to make us super confident that the new logger can replace n-logger and n-mask-logger.

In order to be sure of this, we need to install both of these old modules as development dependencies, then this test script will spin up a Node.js process and log the same data with both the old loggers and the new logger. It then reads the JSON logs _from `stdout`_ and compares them.

This means that we're verifying that, given the same input, n-logger vs reliability-kit/logger and n-mask-logger vs reliability-kit/logger output the same thing. You can see the actual comparisons in the `compatibility-test-cases.js` file, e.g. [here](https://github.com/Financial-Times/dotcom-reliability-kit/blob/21c9079bdc633b95ce3ce21e735081d09068ccb0/packages/logger/test/end-to-end/compatibility-test-cases.js#L33-L44).

```js
[
	{
		id: 'n-logger-docs-1',
		description: '.log() with log level and message (n-logger docs)',

                // This defines that call that is made to each logger. This means that the logger will
                // be instantiated and have the `log` method called with `info` and `Saying hello`
		call: {
			method: 'log',
			args: ['info', 'Saying hello']
		},

                // This describes the output that is expected from each logger. We define them
                // separately because there are documented cases where they differ slightly and
                // we're OK with it.
		expectedOutput: {
			nextLogger: { level: 'info', message: 'Saying hello' },
			reliabilityKit: { level: 'info', message: 'Saying hello' }
		}
	}
]
```

If you want to chat more about it I'd be happy to go through in person 🙂 I'm quite proud of the way this works and it's made me pretty confident that a direct replacement will work for 90% of use cases.

Reply:
Gonna merge this as you said it's non-blocking. I'm kind of intending on writing a blog post about the testing process when we go more public with the experimental logger so will use that to explain further if necessary